### ISSUE TITLE
CRDs listed in IgnoredCRDList are causing RBAC errors in cnrm-resource-stats-recorder

### ISSUE THREAD

**Issue by dwilliams782** [GoogleCloudPlatform/k8s-config-connector#2874]
### Checklist

- [X] I did not find a related open issue.
- [X] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [X] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Bug Description

Apologies if this is a duplicate but I couldn't find it.

We've upgraded to 1.123.1, and we get the following RBAC logs in cnrm-resource-stats-recorder:

```
E1008 12:19:54.881132       1 main.go:183] "error recording metrics for CRD" err="error listing objects for gameservices.cnrm.cloud.google.com/v1beta1, Kind=GameServicesRealm: error listing objects: gameservicesrealms.gameservices.cnrm.cloud.google.com is forbidden: User \"system:serviceaccount:cnrm-system:cnrm-resource-stats-recorder\" cannot list resource \"gameservicesrealms\" in API group \"gameservices.cnrm.cloud.google.com\" at the cluster scope" gvk="gameservices.cnrm.cloud.google.com/v1beta1, Kind=GameServicesRealm"

E1008 12:19:54.881242       1 main.go:183] "error recording metrics for CRD" err="error listing objects for core.cnrm.cloud.google.com/v1alpha1, Kind=ServiceMapping: error listing objects: servicemappings.core.cnrm.cloud.google.com is forbidden: User \"system:serviceaccount:cnrm-system:cnrm-resource-stats-recorder\" cannot list resource \"servicemappings\" in API group \"core.cnrm.cloud.google.com\" at the cluster scope" gvk="core.cnrm.cloud.google.com/v1alpha1, Kind=ServiceMapping"
```

Both of these CRDs are listed in [IgnoredCRDList](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/operator/pkg/k8s/constants.go#L66C2-L66C16), but we still get the RBAC errors as above. I've since deleted these CRDs from the cluster to silence the logs, but I wanted to share for awareness.

### Additional Diagnostic Information

n/a

### Kubernetes Cluster Version

v1.30.3-gke.1969001

### Config Connector Version

1.123.1

### Config Connector Mode

cluster mode

### Steps to reproduce the issue

Check the logs for cnrm-resource-stats-recorder with ignored CRDs installed on the cluster.

**Comment by sunday-ayandokun-exa** [comment]
We are also seeing the same issue after upgrading to v1.123.1, the bug is also present in v1.124.1
```log
E1018 20:01:10.979046       1 main.go:183] "error recording metrics for CRD" err="error listing objects for gameservices.cnrm.cloud.google.com/v1beta1, Kind=GameServicesRealm: error listing objects: gameservicesrealms.gameservices.cnrm.cloud.google.com is forbidden: User \"system:serviceaccount:cnrm-system:cnrm-resource-stats-recorder\" cannot list resource \"gameservicesrealms\" in API group \"gameservices.cnrm.cloud.google.com\" at the cluster scope" gvk="gameservices.cnrm.cloud.google.com/v1beta1, Kind=GameServicesRealm"
```

**Comment by yuwenma** [comment]
Thank you for reporting! We are looking into it and will update later.

**Comment by maqiuyujoyce** [comment]
Hi @dwilliams782 @sunday-ayandokun-exa , thank you for reporting the issue.

To work around the issue on you described, you can run the following commands to remove the no-longer-supported CRDs:

```
kubectl delete crd servicemappings.core.cnrm.cloud.google.com
kubectl delete crd gameservicesrealms.gameservices.cnrm.cloud.google.com
```

Note that the two problematic CRDs have been deleted from the supported CRD suite a while ago so deleting them on your cluster should not cause any issues: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/v1.124.0/operator/pkg/k8s/constants.go#L66. Config Connector doesn't do CRD cleanup, that's why they are still in your cluster.

I believe this reported issue was introduced by https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2631 and here is the fix: #2972.
